avalon/maya/pipeline.py

```diff
@@ -35,6 +35,11 @@ def install(config):
     _register_callbacks()
     _set_project()
 
+    # Check if maya version is compatible else fix it, Maya2018 only
+    # Should be run regardless of batch mode
+    if hasattr(cmds, "about") and cmds.about(version=True) == "2018":
+        compat.install()
```

Comment:
I wouldn't expect this to call `compat.install()`, but I'd expect it to call `compat.remove_googleapiclient` so the code here explains what it does. 

The `install()` in `compat` module for me seems redundant. Unless that `compat.install()` does the check internally in that file. Whether that's preferred maybe @mottosso might have some better arguments about that.

Reply:
Yeah, we chatted about that in private (for some reason?). I'd prefer `compat.install()` as I expect there to be more compatibility fixes long term and I'd rather not pollute the `pipeline.install` logic and readability with it.

Reply:
Ah ok, so then the compatibility checks are moved there as well. (version checks, etc.) And `compat.install()` is just triggered here. That makes sense to me.

Reply:
I 've moved the version check to the `compat` module. Made more sense to me and less contamination of the `pipeline.install`.